python/src/aac/plugins/lsp_server/providers/symbols.py

```diff
@@ -0,0 +1,80 @@
+"""Module contain common functionality shared by the various LSP providers."""
+from enum import Enum
+
+from aac.lang.language_context import LanguageContext
+
+
+class SymbolType(Enum):
+    """Enums classifying the symbol type. Leveraged by providers to tailor responses."""
+    DEFINITION_NAME = 1
+    ENUM_VALUE_TYPE = 2
+    ROOT_KEY_NAME = 3
+
+
+def get_symbol_type(name: str, language_context: LanguageContext) -> list[SymbolType]:
+    """
+    Return a list of enums indicating what kind of symbol this may be based on the language context.
+
+    Args:
+        name (str): The name or symbol to lookup
+        language_context (LanguageContext): The language context to search for context for the symbol.
+
+    Returns:
+        A list of enums indicating a relationship between the name and content in the language context.
+    """
+
+    symbol_types = []
+
+    if language_context.is_definition_type(name):
+        symbol_types.append(SymbolType.DEFINITION_NAME)
+
+    enum_values_in_context = [value for values in [definition.get_top_level_fields().get("values") for definition in language_context.get_definitions_by_root_key("enum")] for value in values]
```

Comment:
I'd certainly be down to see this added to the LanguageContext. 

The for in for loop is an attempt to replace the use of flatten ([because itertools is written in c](https://github.com/python/cpython/blob/main/Modules/itertoolsmodule.c)) with a pythonic alternative, but I can certainly prioritize the readability in a language context function.